tests/generator/base.py

```diff
@@ -277,8 +277,6 @@ def validate_generated_yaml(self, yaml_input):
             if os.path.isfile(output_path):
                 with open(output_path, 'r') as generated:
                     output_yaml = yaml.safe_load(generated.read())
-            else:
-                output_yaml = yaml.safe_load('')
```

Comment:
This change seems kind of unrelated. Could you please clarify why it is part of this commit?

Reply:
I had to dig into my memory to remember why I changed this one, and I must have been tired when I decided that it would be the good solution: I removed those lines because they weren't hit by code coverage anymore. I didn't bother wondering why, I just removed them.

Now that I've actually thought about it, it means there's a subtle regression: `write_netplan_conf_full()` will now create an empty file if the global state is empty, whereas the previous version wouldn't touch the FS.

There are multiple options there:
1/ we simply don't care
2/ we approximate the previous behavior by removing the created file if it's empty
3/ we keep the exact behavior as before by creating a new function
```c
gboolean
netplan_state_nondefault_globals(NetplanState *np_state)
{
    return (np_state->backend != NETPLAN_BACKEND_NONE)
        || has_openvswitch(&np_state->ovs_settings, NETPLAN_BACKEND_NONE, NULL);
}
```
and using it in `write_netplan_conf_full` to do an early check 

```c
if (!netplan_state_nondefault_globals(&global_state) || !netplan_state_get_netdefs_size(&global_state))
    return;
```

Now that I've done the work, the latter approach is nicer :)

Reply:
Addressed in the latest patchset.

Reply:
Empty files can cause issues (e.g. with snapd get/set, as addressed with the most recent netplan.io SRUs in Ubuntu), so thank you for going with option (3) and keeping status quo!